src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -236,18 +282,12 @@ private void OnDiagnosticsUpdatedOnForeground(DiagnosticsUpdatedArgs e)
                     return;
                 }
 
+                _shutdownCancellationSource.Token.ThrowIfCancellationRequested();
```

Comment:
you're on the UI thread.  if you throw an exception, what happens?  what is higher in the stack to catch this?

Reply:
this runs in foreground notification service, it will be caught and ignored.

Reply:
(I think I've already caught bugs masked by that foreground notification service behavior. Would prefer exceptions just took down process, cancellation included, if exceptions leak that far.)

Reply:
so, should I throw or just return if cancelled?